wasi/wasi.go

```diff
@@ -154,6 +172,17 @@ const (
 	wasiSnapshotPreview1Name = "wasi_snapshot_preview1"
 )
 
+type RandomSource interface {
+	Read([]byte) (int, error)
+}
+
+// Non-deterministic random source using crypto/rand
+type CryptoRandomSource struct{}
```

Comment:
Same

Reply:
We do the same for clock so could you reference that and please make the aligned